cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -695,6 +696,42 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 	return nil
 }
 
+func (a *StaticAutoscaler) addUpcomingNodesToClusterSnapshot(upcomingCounts map[string]int, nodeInfosForGroups map[string]*schedulerframework.NodeInfo) error {
+	nodeGroups := a.nodeGroupsById()
+	upcomingNodeGroups := make(map[string]int)
+	upcomingNodesFromUpcomingNodeGroups := 0
+	for nodeGroupName, upcomingNodes := range getUpcomingNodeInfos(upcomingCounts, nodeInfosForGroups) {
+		nodeGroup := nodeGroups[nodeGroupName]
+		if nodeGroup == nil {
+			return fmt.Errorf("failed to find node group: %s", nodeGroupName)
+		}
+		for _, upcomingNode := range upcomingNodes {
+			var pods []*apiv1.Pod
+			for _, podInfo := range upcomingNode.Pods {
+				pods = append(pods, podInfo.Pod)
+			}
+			err := a.ClusterSnapshot.AddNodeWithPods(upcomingNode.Node(), pods)
+			if err != nil {
+				klog.Errorf("Failed to add upcoming node %s to cluster snapshot: %v", upcomingNode.Node().Name, err)
```

Comment:
`err` will be logged outside of this function as well, so I'd skip logging the error here and instead of:

```
klog.Errorf("Failed to add upcoming node %s to cluster snapshot: %v", upcomingNode.Node().Name, err)
return err
```

just do 

```
return fmt.Errorf("Failed to add upcoming node %s to cluster snapshot: %w", upcomingNode.Node().Name, err)
```